service/src/main/java/uk/nhs/adaptors/gp2gp/ehr/mapper/EncounterComponentsMapper.java

```diff
@@ -0,0 +1,125 @@
+package uk.nhs.adaptors.gp2gp.ehr.mapper;
+
+import java.util.Arrays;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.stream.Collectors;
+
+import org.apache.commons.lang3.StringUtils;
+import org.hl7.fhir.dstu3.model.CodeableConcept;
+import org.hl7.fhir.dstu3.model.Condition;
+import org.hl7.fhir.dstu3.model.Encounter;
+import org.hl7.fhir.dstu3.model.Immunization;
+import org.hl7.fhir.dstu3.model.ListResource;
+import org.hl7.fhir.dstu3.model.Observation;
+import org.hl7.fhir.dstu3.model.ProcedureRequest;
+import org.hl7.fhir.dstu3.model.Resource;
+import org.hl7.fhir.dstu3.model.ResourceType;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import com.google.common.collect.ImmutableMap;
+
+import lombok.RequiredArgsConstructor;
+
+@Component
+@RequiredArgsConstructor(onConstructor = @__(@Autowired))
+public class EncounterComponentsMapper {
+
+    private static final String CONSULTATION_LIST_CODE = "325851000000107";
+    private static final String TOPIC_LIST_CODE = "25851000000105";
+    private static final String CATEGORY_LIST_CODE = "24781000000107";
+    private static final List<String> COMPONENTS_LISTS = Arrays.asList(TOPIC_LIST_CODE, CATEGORY_LIST_CODE);
+
+    private static final String NARRATIVE_STATEMENT_CODE = "37331000000100";
+    private static final String NOT_IMPLEMENTED_MAPPER_PLACE_HOLDER = "<!-- %s/%s -->";
+    private static final boolean IS_NESTED = false;
+
+    private final Map<ResourceType, Function<Resource, String>> encounterComponents = ImmutableMap.<ResourceType, Function<Resource,
+        String>>builder()
+        .put(ResourceType.List, this::mapListResource)
+        .put(ResourceType.ProcedureRequest, this::mapProcedureRequest)
+        .put(ResourceType.Observation, this::mapObservation)
+        .put(ResourceType.Immunization, this::mapImmunization)
+        .put(ResourceType.Condition, this::mapCondition)
+        .build();
+
+    private final MessageContext messageContext;
+
+    private final DiaryPlanStatementMapper diaryPlanStatementMapper;
+
+    private final NarrativeStatementMapper narrativeStatementMapper;
+    private final ObservationStatementMapper observationStatementMapper;
+    private final ImmunizationObservationStatementMapper immunizationObservationStatementMapper;
+    private final ConditionLinkSetMapper conditionLinkSetMapper;
+
+    public String mapComponents(Encounter encounter) {
+        Optional<ListResource> listReferencedToEncounter =
+            messageContext.getInputBundleHolder().getListReferencedToEncounter(encounter.getIdElement(), CONSULTATION_LIST_CODE);
+
+        return listReferencedToEncounter
+            .map(this::mapItemsTpComponents)
+            .orElse(StringUtils.EMPTY);
+    }
+
+    private String mapItemsTpComponents(ListResource listReferencedToEncounter) {
```

Comment:
typo ("Tp") and different method name suggestion to not be confused with the next method
```suggestion
    private String mapListResourceToComponents(ListResource listReferencedToEncounter) {
```